src/proxyServer.ts

```diff
@@ -1,101 +0,0 @@
-import cors from "@koa/cors";
-import Koa from "koa";
-import proxy from "koa-proxies";
-import { CopilotSettings } from "@/settings/SettingsPage";
-import { ChatModelDisplayNames } from "@/constants";
-
-// There should only be 1 running proxy server at a time so keep it in upper scope
-let server: any;
-
-export class ProxyServer {
```

Comment:
@logancyang I'm wondering if we can remove this ProxyServer entirely and just make calls directly to the right backend. What was the original reason for having this proxy in the first place?

Reply:
@gianluca-venturini this is for some model providers such as Claude and all those 3rd party OpenAI replacements who don't allow CORS. Without this workaround they result in CORS errors in Obsidian.

Reply:
Thanks @logancyang for adding some context here. I found out that we can use the function [requestUrl()](https://docs.obsidian.md/Reference/TypeScript+API/requestUrl) instead of implementing a Proxy, I've an initial implementation here [acd0481](https://github.com/logancyang/obsidian-copilot/pull/513/commits/acd04819e26a02241adf333dba6d5f11facf2dbb). For now I have it working with `ChatAnthropic`, but we can easily replace all the calls made inside the different models.